src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt

```diff
@@ -119,15 +119,35 @@ class WalletControllerTest {
             .expectStatus()
             .isCreated
             .expectBody()
-            .json(
-                objectMapper.writeValueAsString(
-                    WalletCreateResponseDto()
-                        .walletId(WALLET_DOMAIN.id.value)
-                        .redirectUrl(
-                            "$webviewPaymentUrl#walletId=${WALLET_DOMAIN.id.value}&useDiagnosticTracing=${WalletTestUtils.CREATE_WALLET_REQUEST.useDiagnosticTracing}&paymentMethodId=${WalletTestUtils.CREATE_WALLET_REQUEST.paymentMethodId}"
-                        )
-                )
-            )
+            .jsonPath("$.walletId")
+            .value<String> { walletId ->
+                // Assert that the walletId is as expected
+                assert(walletId.startsWith(WALLET_DOMAIN.id.value.toString().trim())) {
+                    "walletId is not the expected value"
+                }
+            }
+            .jsonPath("$.redirectUrl")
+            .value<String> { redirectUrl ->
+                // Assert that the redirectUrl starts with the expected base URL
+                assert(redirectUrl.contains("#walletId=${WALLET_DOMAIN.id.value}")) {
+                    "Redirect URL does not contains the expected walletId in fragment"
+                }
+                // Check for the presence of other parameters
+                assert(
+                    redirectUrl.contains(
+                        "useDiagnosticTracing=${WalletTestUtils.CREATE_WALLET_REQUEST.useDiagnosticTracing}"
+                    )
+                ) {
+                    "Redirect URL does not contain the expected useDiagnosticTracing parameter"
+                }
+                assert(
+                    redirectUrl.contains(
+                        "paymentMethodId=${WalletTestUtils.CREATE_WALLET_REQUEST.paymentMethodId}"
+                    )
+                ) {
+                    "Redirect URL does not contain the expected paymentMethodId parameter"
+                }
+            }
```

Comment:
we are not checking that returned url contains the newly added query parameter in this test. Please add missing verification here